nats.go

```diff
@@ -4410,6 +4412,65 @@ func (s *Subscription) Drain() error {
 	return conn.unsubscribe(s, 0, true)
 }
 
+type DrainStatus interface {
+	PendingMsgs() int
+	Draining() bool
+	Complete() <-chan struct{}
+}
+
+type drainStatus struct {
+	sub *Subscription
+}
+
+func (s *Subscription) DrainStatus() DrainStatus {
+	s.mu.Lock()
+	defer s.mu.Unlock()
+	return &drainStatus{
+		sub: s,
+	}
+}
+
+func (s *drainStatus) Draining() bool {
+	if s.sub == nil {
+		return false
+	}
+	s.sub.mu.Lock()
+	defer s.sub.mu.Unlock()
+	return s.sub.draining
+}
+
+func (s *drainStatus) Complete() <-chan struct{} {
+	var drainCh chan struct{}
+	if s.sub == nil {
+		// if the subscription is nil, return a closed channel
+		// to avoid blocking on the caller side.
+		drainCh = make(chan struct{})
+		close(drainCh)
+		return drainCh
+	}
+	s.sub.mu.Lock()
+	defer s.sub.mu.Unlock()
+	if s.sub.drainingComplete == nil {
+		drainCh = make(chan struct{})
+		s.sub.drainingComplete = drainCh
```

Comment:
`drainingComplete` maybe too long, could be instead `sub.drainCh` like here or `sub.dch`

Reply:
`drainCh` sounds good, I'll change it. `dch` is a bit too ambiguous for my liking.